tasks/task_manager.cc

```diff
@@ -17,6 +17,73 @@ namespace tasks {
 
 logging::logger tmlogger("task_manager");
 
+bool task_manager::task::children::all_finished() const noexcept {
+    return _children.empty();
+}
+
+size_t task_manager::task::children::size() const noexcept {
+    return _children.size() + _finished_children.size();
+}
+
+future<> task_manager::task::children::add_child(foreign_task_ptr task) {
+    rwlock::holder exclusive_holder = co_await _lock.hold_write_lock();
+
+    auto id = task->id();
+    _children.emplace(id, std::move(task));
+}
+
+future<> task_manager::task::children::mark_as_finished(task_id id) const {
+    rwlock::holder exclusive_holder = co_await _lock.hold_write_lock();
+
+    auto& child_ptr = _children.at(id);
+    auto child = co_await smp::submit_to(child_ptr.get_owner_shard(), [&child_ptr] () -> future<task_essentials> {
+        co_return task_essentials{
+            .task_status = child_ptr->get_status(),
+            .task_progress = co_await child_ptr->get_progress(),
+            .parent_id = child_ptr->get_parent_id(),
+            .type = child_ptr->type(),
+            .abortable = child_ptr->is_abortable(),
+            .failed_children = co_await child_ptr->get_failed_children(),
+        };
+    });
+    _finished_children.push_back(child);
```

Comment:
Can we make _finished_children an intrusive list so we won't have to allocate anything, and just hang the task_essentials object onto it?

Reply:
What would keep them alive then?